### PR TITLE
Exit successfully for a successful bundle oudated run

### PR DESCRIPTION
**Pull request by ngan**
## What was the end-user or developer problem that led to this PR?

After [this change](https://github.com/rubygems/rubygems/pull/7966), we're finding programmatically impossible to determine if running `bundle outdated --parseable` was successful or not. This command `exit 1` when there are dependencies that are out of date:

https://github.com/rubygems/rubygems/blob/c63c4b0305de7a5712436d37decca648b14b0091/bundler/lib/bundler/cli/outdated.rb#L122

Before #7966, we were scanning `stderr` to a non-empty string to determine if the command ran successful. I'm not sure what's the best way to do that now.

I think `bundle outdated` should exit 0 (cleanly) on a successful run, **whether there are outdated dependencies or not**. I'd be very surprised if there are projects out there that use this command programmatically with 0 out of date dependencies. Even then, folks could parse `stdout` for empty string (if using `--parseable`). Looking at the specs, `bundle oudated` is expected to `exit 1` when there's actual errors (eg invalid gem, or parsing errors, etc) so I think this is overloading `exit 1`.

## What is your fix for the problem, implemented in this PR?

Remove the `exit 1`.

## Make sure the following tasks are checked

- [x] Describe the problem / feature
- [x] Write [tests](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#tests) for features and bug fixes
- [x] Write code to solve the problem
- [x] Make sure you follow the [current code style](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#code-formatting) and [write meaningful commit messages without tags](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#commit-messages)

### CODE REVIEW

**Comment by deivid-rodriguez** [comment]
Ooops, we specifically discussed backwards compatibility of that particular change as a risk. @mattbrictson mentioned your exact use case at https://github.com/rubygems/rubygems/issues/7938#issuecomment-2293699455, and I acknowledged it as risky, but then we ended up doing it without further discussion 🙈.

In any case, I don't think the following is correct:

> Looking at the specs, `bundle oudated` is expected to `exit 1` when there's actual errors (eg invalid gem, or parsing errors, etc) so I think this is overloading `exit 1`.

As I mentioned in https://github.com/rubygems/rubygems/issues/7938#issuecomment-2293728427, Bundler should be exiting with different status codes for more generic errors like the one you mentioned.

For example, I tried `bundle outdated gem-not-found` and it exits with status code 7.

To sum up:

* I would not introduce this PR because it's changing behavior people may be relying on.
* I think #7966 was a good change but we need to reevaluate it's backwards compatibility, depending on how many people are affected. We may want to revert it and introduce it in Bundler 2.6 as a small breaking change.
* We need to document status codes, that the error range for "general errors" is status_code >= 2), and that each commands decides when to return exit code 0 or 1 according to its own semantics. And also advice to not rely on empty stderr for command success.

**Comment by ngan** [comment]
I see. Maybe I can check for an exit code > 1 for cases that something went wrong?

**Comment by deivid-rodriguez** [comment]
Yes, that's my suggestion and what I plan to document 👍.
